posix/consolefs.c

```diff
@@ -0,0 +1,398 @@
+        OE_RAISE_ERRNO(OE_ENOMEM);
+
+    ret = _consolefs_write(desc, buf, buf_size);
+
+done:
+
+    if (buf)
+        oe_free(buf);
+
+    return ret;
+}
+
+static oe_host_fd_t _consolefs_gethostfd(oe_fd_t* file_)
+{
+    oe_host_fd_t ret = -1;
+    file_t* file = _cast_file(file_);
+
+    if (!file)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    ret = file->host_fd;
+
+done:
+    return ret;
+}
+
+static oe_off_t _consolefs_lseek(oe_fd_t* file_, oe_off_t offset, int whence)
+{
+    oe_off_t ret = -1;
+
+    OE_UNUSED(file_);
+    OE_UNUSED(offset);
+    OE_UNUSED(whence);
+    OE_RAISE_ERRNO(OE_ESPIPE);
+
+done:
+    return ret;
+}
+
+static int _consolefs_close(oe_fd_t* file_)
+{
+    int ret = -1;
+    file_t* file = _cast_file(file_);
+
+    if (!file)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    /* Ask the host to perform this operation. */
+    {
+        if (oe_posix_close_ocall(&ret, file->host_fd) != OE_OK)
+            OE_RAISE_ERRNO(OE_EINVAL);
+
+        if (ret == -1)
+            OE_RAISE_ERRNO(oe_errno);
+    }
+
+    /* Free the file structure. */
+    oe_free(file);
+
+done:
+    return ret;
+}
+
+static int _consolefs_getdents64(
+    oe_fd_t* file,
+    struct oe_dirent* dirp,
+    uint32_t count)
+{
+    OE_UNUSED(file);
+    OE_UNUSED(dirp);
+    OE_UNUSED(count);
+
+    /* The standard devices are not directories, so this is unsupported. */
+    OE_RAISE_ERRNO(OE_ENOTSUP);
+
+done:
+    return -1;
+}
+
+static oe_file_ops_t _ops = {
+    .fd.read = _consolefs_read,
+    .fd.write = _consolefs_write,
+    .fd.readv = _consolefs_readv,
+    .fd.writev = _consolefs_writev,
+    .fd.dup = _consolefs_dup,
+    .fd.ioctl = _consolefs_ioctl,
+    .fd.fcntl = _consolefs_fcntl,
+    .fd.close = _consolefs_close,
+    .fd.get_host_fd = _consolefs_gethostfd,
+    .lseek = _consolefs_lseek,
+    .getdents64 = _consolefs_getdents64,
+};
+
+static oe_file_ops_t _get_ops(void)
+{
+    return _ops;
+}
+
+static oe_fd_t* _new_file(uint32_t fileno)
+{
+    oe_fd_t* ret = NULL;
+    file_t* file = NULL;
+    struct _args
+    {
+        const char* path;
+        int flags;
+    };
+    struct _args args[] = {
+        {"/dev/stdin", OE_O_RDONLY},
+        {"/dev/stdout", OE_O_WRONLY},
+        {"/dev/stderr", OE_O_WRONLY},
+    };
+
+    if (fileno > OE_STDERR_FILENO)
+        goto done;
+
+    /* Create the file struct. */
+    {
+        if (!(file = oe_calloc(1, sizeof(file_t))))
+            goto done;
+
+        file->base.type = OE_FD_TYPE_FILE;
+        file->base.ops.file = _ops;
+        file->magic = MAGIC;
+    }
+
+    /* Ask the host to open this file. */
+    {
+        oe_host_fd_t retval;
+        const char* path = args[fileno].path;
+        int flags = args[fileno].flags;
+
+        if (oe_posix_open_ocall(&retval, path, flags, 0) != OE_OK)
+            goto done;
+
+        if (retval < 0)
+            goto done;
+
+        oe_printf("STANDARD_DEVICE=%ld\n", retval);
```

Comment:
>oe_printf("STANDARD_DEVICE=%ld\n", retval); [](start = 8, length = 43)

Is this a debugging statement that should be removed? Otherwise, consider `OE_TRACE_MSG`

Reply:
Yep. Removed.